index.php

```diff
@@ -20,9 +20,14 @@
  * Note: TODO comments mark missing and incomplete code.
  */
 
-// Read configuration for database access.
-require_once 'config.php';
+// Read configuration
+$confdir = 'local-template';
+if (is_dir('local')) {
```

Comment:
This will only work if `index.php` is called from the same directory which also includes the `local` directory.

Reply:
Maybe just using `include_once` for `local/config.php` and checking the result could work. Then `local-template` could be included if an error occurred.

Is `local/config.php` a good name? After all, it is searched at different locations. So if that rather generic name happens to be found somewhere else, that would cause undefined behaviour. Using `./local/config.php` would ignore the include path, so that would be better.

Reply:
Good points... Would something like this be acceptable instead?
```
$scriptdir = dirname(__FILE__);
$confdir = $scriptdir . '/local-template';
if (is_dir($scriptdir . '/local')) {
    $confdir = $scriptdir . '/local';
}
require_once "$confdir/config.php";
```

Reply:
Using the absolute path from makes sense